src/test/java/homeworks/POM/Pages/BasePageHmw.java

```diff
@@ -13,14 +14,35 @@ public class BasePageHmw {
     WebDriverWait wait;
     Actions actions;
     By allSongsMenuItemLocator =By.cssSelector(" li a.songs");
+    By avatarLocator = By.cssSelector("img.avatar");
+    By soundBarPlayLocator = By.cssSelector("[data-testid = 'sound-bar-play']");
 
     public BasePageHmw(WebDriver givenDriver){
         driver= givenDriver;
         wait =new WebDriverWait(driver, Duration.ofSeconds(5));
+        actions =new Actions(driver);
+        PageFactory.initElements(driver,this);
+    }
+    public WebElement getUserAvatar(){
+        return driver.findElement(avatarLocator);
+    }
+
+    public boolean isUserAvatarDisplayed(){
+        return driver.findElement(avatarLocator).isDisplayed();
     }
 
-    public void clickOnAllSongs(){
-      driver.findElement(allSongsMenuItemLocator).click();
+    public boolean isSongPlaying() {
+        WebElement soundBarVisualizer = driver.findElement(soundBarPlayLocator);
+        return soundBarVisualizer.isDisplayed();
+    }
+    public  AllSongsPageHmw clickOnAllSongs(){
+        driver.findElement(allSongsMenuItemLocator).click();
+        return new AllSongsPageHmw(driver);
     }
+//    public  WebElement hoverPlay() {
+//        WebElement play=driver.findElement(By.cssSelector("[data-testid='play-btn']"));
+//        actions.moveToElement(play).perform();
+//        return play;
```

Comment:
**Base Page \ Base Class** — Implements the necessary methods for work with webdriver.

We will store the web elements present on each page in the corresponding java class.

The methods `getUserAvatar ()`,isUserAvatarDisplayed()  must be stored in the `class HomePage` because the avatar is visible on the web site -home page

The methods `isSongPlaying() `, `clickOnAllSongs()` must be stored in the  `class AllSongsPage`  because the song is visible on the web site -songs page